smartcontractkit/external-adapters-js
# Add API Key support for Coinpaprika

## Closes 24006

## Description

Allow NOPs to start the Coinpaprika adapter with an API KEY.  This is required as Coinpaprika will soon be severely limiting their free tier.

## Changes

- Support setting an `API_KEY` environment variable for Coinpaprika
- Support setting a `IS_TEST_MODE` environment variable for Coinpaprika.  This is required for now as this need to be passed in the header for the API key to have an effect whilst the Coinpaprika API is in the testing phase.  This can be removed later on once their API is in production.
- Add unit tests to verify the correct adapter configuration

## Steps to Test

1. Set `API_KEY` and `IS_TEST_MODE` and send a request to the adapter to verify that it is working.
2. Unset both environment variables and ensure that the adapter is still working.

## Quality Assurance
